src/extensions/apollo/index.ts

```diff
@@ -0,0 +1,21 @@
+import { jsonToGraphQLQuery } from '../../index';
+
+export function addApolloTargetToGqlQueryString(query: string, target: string): string {
+    for (let i = query.length; i--; i === 0) {
+        const thisChar = query[i];
+        if ((thisChar !== ' ') && (thisChar !== '}')) {
+            return query.slice(0, i + 1) + ' @' + target + query.slice(i + 1 , query.length);
+        }
+    }
+}
+
+export function objToApolloQuery(object: object, target: string): string {
+    const gqlQuery = jsonToGraphQLQuery(object);
+    const querySansTypenames = gqlQuery.replace(/__typename /g, '');
+    const queryWithTarget = addApolloTargetToGqlQueryString(querySansTypenames, target);
+    return 'query { ' + queryWithTarget + ' }';
+}
```

Comment:
Rather than doing string replacement, I would prefer we embed this information in the query object itself.

I've created an issue to track the implementation of "Directives" which is basically what we're doing: https://github.com/dupski/json-to-graphql-query/issues/7

Looking at the usage of `@client` from https://github.com/apollographql/apollo-link-state

How about something like this:

```ts
const query = {
  mutation: {
    updateNetworkStatus: {
      __args: {
        isConnected: true
      },
      __directives: {
        client: true
      }
    }
  }
};
```

This would then result in:

```graphql
mutation {
    updateNetworkStatus(isConnected: true) @client
}
```

That will work for mutations (or rather nodes where sub-nodes are selected), but in the case of fields, we'll need something slightly different. Maybe:

```ts
const query = {
  Posts: {
    id: true,
    name: true,
    author: new WithDirectives({
      client: true
    })
  }
};
```

Which will give us:

```graphql
query {
  Posts {
    id
    name
    author @client
  }
}
```

If you have a look at how we've implemented `__alias` and `EnumType` in the code, that should give you some pointers on how to implement this :)

Reply:
Got it. I'm on it~

Reply:
Actually, I have some thoughts on this. I think I'll continue the conversation in issue #7 though.